subsys/net/lib/nrf_cloud/src/nrf_cloud_codec.c

```diff
@@ -889,20 +859,283 @@ static int nrf_cloud_encode_service_info_ui(const struct nrf_cloud_svc_info_ui *
 	return 0;
 }
 
+#ifdef CONFIG_MODEM_INFO
+
+#define MODEM_INFO_JSON_KEY_NET_INF	"networkInfo"
+#define MODEM_INFO_JSON_KEY_SIM_INF	"simInfo"
+#define MODEM_INFO_JSON_KEY_DEV_INF	"deviceInfo"
+
+static int add_modem_info_data(struct lte_param *param, cJSON *json_obj)
+{
+	char data_name[MODEM_INFO_MAX_RESPONSE_SIZE];
+	enum at_param_type data_type;
+	int ret;
+
+	__ASSERT_NO_MSG(param != NULL);
+	__ASSERT_NO_MSG(json_obj != NULL);
+
+	memset(data_name, 0, ARRAY_SIZE(data_name));
+	ret = modem_info_name_get(param->type,
+				data_name);
+	if (ret < 0) {
+		LOG_DBG("Data name not obtained: %d", ret);
+		return -EINVAL;
+	}
+
+	data_type = modem_info_type_get(param->type);
+	if (data_type < 0) {
+		return -EINVAL;
+	}
+
+	if (data_type == AT_PARAM_TYPE_STRING &&
+	    param->type != MODEM_INFO_AREA_CODE) {
+		if (cJSON_AddStringToObject(json_obj, data_name, param->value_string) == NULL) {
+			return -ENOMEM;
+		}
+	} else {
+		if (cJSON_AddNumberToObject(json_obj, data_name, param->value) == NULL) {
+			return -ENOMEM;
+		}
+	}
+
+	return ret;
+}
+
+static int encode_modem_info_network(struct network_param *network, cJSON *json_obj)
+{
+	char network_mode[12] = {0};
+	char data_name[MODEM_INFO_MAX_RESPONSE_SIZE] = {0};
+	int ret;
+
+	__ASSERT_NO_MSG(network != NULL);
+	__ASSERT_NO_MSG(json_obj != NULL);
+
+	ret = add_modem_info_data(&network->current_band, json_obj);
```

Comment:
Rather than call `add_modem_info_data`, which has to then look up `modem_info_name_get` and `modem_info_type_get`, and then which just immediately calls `cJSON_AddStringToObject` with the data found in that lookup,

Why not just call `cJSON_AddStringToObject` directly here with the correct values (taken from `#define`s).

That would line up more with the style of how the rest of the cloud codec works, and I think would be easier to follow

If you do this, both `modem_info_type_get` and `modem_info_name_get` could be completely removed.

Reply:
Tagging @jayteemo again, here, to make sure he agrees this makes sense

Reply:
i do not have an opinion on this, as it may increase the scope of the PR.